This notebook was used to tidy up the extracted data. This involves checking that ids match, removing unneeded columns and stacking together separate indices into a single data frame.

# 1. Extracted S2-indices for 2017 (DONE)

- tidy up the data
- add "id_17" index

In [1]:
import pandas as pd

In [2]:
%%time

src_pth = f".\\csvs\\ZV2017_extracted_ndvi.csv"

df0 = pd.read_csv(src_pth)

Wall time: 3.13 s


In [3]:
df0.tail()

Unnamed: 0  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS  \
664133      664133   1892284     2976726     1300   8760.9728         204   
664134      664134   3588000     2976727     1300   7807.7304         204   
664135      664135   1646627     2976728     1300   7559.2166         204   
664136      664136   3588612     2976729     1300   6120.1665         204   
664137      664137   3588639     2976730     1300   5015.4655         204   

       OBRAZEC  INTERSECT    POVR_ar  REGIJA  ...  p_17-10-08  p_17-10-18  \
664133      GP          0  87.609728       2  ...         NaN         NaN   
664134      GP          0  78.077304       2  ...         NaN         NaN   
664135      GP          0  75.592166       2  ...         NaN         NaN   
664136      GP          0  61.201665       2  ...         NaN         NaN   
664137      GP          0  50.154655       2  ...         NaN         NaN   

        pc075_m02  pc075_m04  pc075_m05  pc075_m06  pc075_m07  pc075_m08  \
664133   0.548966   0.871466   0.835215   0.883261   0.643689   0.814480   
664134   0.541856   0.880461   0.788361   0.884401   0.791368   0.774565   
664135   0.445272   0.819034   0.873336   0.664987   0.590871   0.608201   
664136   0.573511   0.879573   0.900252   0.865085   0.645182   0.663612   
664137   0.625246   0.913037   0.901425   0.884859   0.669561   0.764977   

        pc075_m09  pc075_m10  
664133   0.814480   0.814480  
664134   0.774565   0.774565  
664135   0.608201   0.608201  
664136   0.663612   0.663612  
664137   0.764977   0.764977  

[5 rows x 38 columns]

In [4]:
suffs = ["evi2", "ndvi", "ndwi", "savi"]
for suff in suffs:
    # Open file
    src_pth = f".\\csvs\\ZV2017_extracted_{suff}.csv"
    df0 = pd.read_csv(src_pth)
    
    # Drop unnecessary columns
    df_tidy = df0.drop(columns=df0.iloc[:, 0:12])  # KMRS data
    df_tidy = df_tidy.drop(columns=df0.iloc[:, -8:])  # Monthly values

    # Rename "p_17" to "ndvi" (and other indices)
    df_tidy.columns = df_tidy.columns.str.replace("p_17-", f"{suff}_")

    # Add the "id_17" index column
    df_tidy = df_tidy.reset_index()
    df_tidy = df_tidy.rename(columns={"index": "id_17"})

    print(df_tidy.tail(1))
    
    # Save to csv
    df_tidy.to_csv(f".\\results_extracted\\ZV2017_extracted_{suff}.csv", index=False)

         id_17  evi2_04-11  evi2_04-21  evi2_05-01  evi2_05-11  evi2_05-21  \
664137  664137         NaN    0.779873         NaN         NaN    0.827327   

        evi2_05-31  evi2_06-10  evi2_06-20  evi2_06-30  evi2_07-10  \
664137         NaN    0.751983         NaN         NaN    0.429898   

        evi2_07-20  evi2_07-30  evi2_08-09  evi2_08-19  evi2_08-29  \
664137    0.449394         NaN    0.579503         NaN    0.520069   

        evi2_09-08  evi2_10-08  evi2_10-18  
664137         NaN         NaN         NaN  
         id_17  ndvi_04-11  ndvi_04-21  ndvi_05-01  ndvi_05-11  ndvi_05-21  \
664137  664137         NaN    0.913037         NaN         NaN    0.901425   

        ndvi_05-31  ndvi_06-10  ndvi_06-20  ndvi_06-30  ndvi_07-10  \
664137         NaN    0.884859         NaN         NaN     0.56709   

        ndvi_07-20  ndvi_07-30  ndvi_08-09  ndvi_08-19  ndvi_08-29  \
664137    0.772031         NaN    0.749255         NaN    0.780699   

        ndvi_09-08  ndvi_10-08  

# 2. Extracted S2-rgbn for 2017 (DONE)

- tidy up the data
- add "id_17" index

In [ ]:
import pandas as pd

In [5]:
%%time

src_pth = f".\\csvs\\ZV2017_extracted_s2-rgbn.csv"

df0 = pd.read_csv(src_pth)

Wall time: 9.82 s


In [13]:
df0

GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS OBRAZEC  \
0        2537988     2104906     1100  19308.9997         801      GP   
1        2537988     2104908     1100  19308.9997           5      GP   
2        2537988     2104910     1100  19308.9997         203      GP   
3        1143420     2105003     1100   8277.3796         801      GP   
4        3077991     2652377     1211  14987.7976         100      GP   
...          ...         ...      ...         ...         ...     ...   
664133   1892284     2976726     1300   8760.9728         204      GP   
664134   3588000     2976727     1300   7807.7304         204      GP   
664135   1646627     2976728     1300   7559.2166         204      GP   
664136   3588612     2976729     1300   6120.1665         204      GP   
664137   3588639     2976730     1300   5015.4655         204      GP   

        INTERSECT     POVR_ar  REGIJA  blue_04_01  ...  red_10_18  nir_10_18  \
0               0   57.930456       5    0.085712  ...        NaN        NaN   
1               0   30.520097       5    0.111312  ...        NaN        NaN   
2               0   26.799442       5    0.082948  ...        NaN        NaN   
3               0   40.952670       5    0.088406  ...        NaN        NaN   
4               0  149.877976       4    0.082925  ...        NaN        NaN   
...           ...         ...     ...         ...  ...        ...        ...   
664133          0   87.609728       2    0.075792  ...        NaN        NaN   
664134          0   78.077304       2    0.090231  ...        NaN        NaN   
664135          0   75.592166       2    0.072073  ...        NaN        NaN   
664136          0   61.201665       2    0.072300  ...        NaN        NaN   
664137          0   50.154655       2    0.063442  ...        NaN        NaN   

        blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
0              NaN          NaN        NaN        NaN         NaN   
1              NaN          NaN        NaN        NaN         NaN   
2              NaN          NaN        NaN        NaN         NaN   
3              NaN          NaN        NaN        NaN         NaN   
4              NaN          NaN        NaN        NaN         NaN   
...            ...          ...        ...        ...         ...   
664133         NaN          NaN        NaN        NaN         NaN   
664134         NaN          NaN        NaN        NaN         NaN   
664135         NaN          NaN        NaN        NaN         NaN   
664136         NaN          NaN        NaN        NaN         NaN   
664137         NaN          NaN        NaN        NaN         NaN   

        green_10_28  red_10_28  nir_10_28  
0               NaN        NaN        NaN  
1               NaN        NaN        NaN  
2               NaN        NaN        NaN  
3               NaN        NaN        NaN  
4               NaN        NaN        NaN  
...             ...        ...        ...  
664133          NaN        NaN        NaN  
664134          NaN        NaN        NaN  
664135          NaN        NaN        NaN  
664136          NaN        NaN        NaN  
664137          NaN        NaN        NaN  

[664138 rows x 137 columns]

In [14]:
# Drop unnecessary columns
df_tidy = df0.drop(columns=df0.iloc[:, 0:9])
   
# Add the "id_17" index column
df_tidy = df_tidy.reset_index()
df_tidy = df_tidy.rename(columns={"index": "id_17"})

df_tidy.tail(3)

id_17  blue_04_01  green_04_01  red_04_01  nir_04_01  blue_04_11  \
664135  664135    0.072073     0.082949   0.077704   0.342441         NaN   
664136  664136    0.072300     0.083633   0.075527   0.311218         NaN   
664137  664137    0.063442     0.075768   0.057569   0.352208         NaN   

        green_04_11  red_04_11  nir_04_11  blue_04_21  ...  red_10_18  \
664135          NaN        NaN        NaN    0.067409  ...        NaN   
664136          NaN        NaN        NaN    0.061263  ...        NaN   
664137          NaN        NaN        NaN    0.059380  ...        NaN   

        nir_10_18  blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
664135        NaN         NaN          NaN        NaN        NaN         NaN   
664136        NaN         NaN          NaN        NaN        NaN         NaN   
664137        NaN         NaN          NaN        NaN        NaN         NaN   

        green_10_28  red_10_28  nir_10_28  
664135          NaN        NaN        NaN  
664136          NaN        NaN        NaN  
664137          NaN        NaN        NaN  

[3 rows x 129 columns]

In [15]:
%%time

# Save to csv
df_tidy.to_csv(f".\\results_extracted\\ZV2017_extracted_s2-rgbn.csv", index=False)

Wall time: 1min 20s


# 3. Extracted S1-slc for 2017 (DONE)

Missing rows ---> SOLVED

Only 664106 instead of 664137, that is 31 missing lines. Maybe investigate what crop types were used in the missing lines!

## 3.1 Find missing values

In [69]:
%%time
import geopandas as gpd

s1_pth = f".\\csvs\\ZV2017_extracted_s1-slc.csv"
s2_pth = f".\\poljine_SHP\\ZV2017_d96tm_clean_ids.shp"

df2 = pd.read_csv(s1_pth)
df1 = gpd.read_file(s2_pth)

Wall time: 2min 17s


In [29]:
df22 = df2[["GERK_PID", "POLJINA_ID"]]
df22.tail(5)

GERK_PID  POLJINA_ID
664101   1892284     2976726
664102   3588000     2976727
664103   1646627     2976728
664104   3588612     2976729
664105   3588639     2976730

In [27]:
df11 = df1[["GERK_PID", "POLJINA_ID"]]
df11.tail(33)

GERK_PID  POLJINA_ID
664105   1283404     3062363
664106    913282     3062364
664107    913286     3062365
664108    907589     3062366
664109    907585     3062367
664110    907587     3062368
664111   3708301     3062370
664112   4617320     3071272
664113   4883395     3071273
664114   4553594     3071274
664115   4553596     3071276
664116   1734856     3071277
664117   1715880     3071279
664118   1734866     3071280
664119   1334252     3071291
664120   3425344     3071292
664121    390270     3072032
664122     57362     3072033
664123     57454     3072034
664124     57361     3072035
664125     57451     3072036
664126     57253     3072037
664127     57368     3072038
664128   4056436     2976721
664129   1628710     2976722
664130   1646610     2976723
664131    421946     2976724
664132   1892289     2976725
664133   1892284     2976726
664134   3588000     2976727
664135   1646627     2976728
664136   3588612     2976729
664137   3588639     2976730

In [36]:
# This is a list of indices from 2017 that are missing from the S1 Data Set
missing_values = pd.concat([df11,df22]).drop_duplicates(keep=False).index
missing_values

Int64Index([ 46616,  47144,  57323,  62652,  62757,  62763,  93813,  94575,
             97683, 109581, 109906, 115507, 116679, 116680, 135561, 154766,
            204863, 260352, 291532, 300465, 326524, 355781, 392932, 429276,
            451339, 530030, 555758, 598878, 649259, 650254, 651497, 656362],
           dtype='int64')

In [70]:
df1.head()

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
0      0   2537988     2104906     1100  19308.9997        801      GP   
1      1   2537988     2104908     1100  19308.9997        005      GP   
2      2   2537988     2104910     1100  19308.9997        203      GP   
3      3   1143420     2105003     1100   8277.3796        801      GP   
4      4   3077991     2652377     1211  14987.7976        100      GP   

   INTERSECT     POVR_ar  REGIJA  \
0          0   57.930456       5   
1          0   30.520097       5   
2          0   26.799442       5   
3          0   40.952670       5   
4          0  149.877976       4   

                                            geometry  
0  POLYGON ((547508.347 86829.736, 547545.426 868...  
1  POLYGON ((547497.850 86785.076, 547578.847 867...  
2  POLYGON ((547599.139 86850.730, 547670.666 868...  
3  POLYGON ((547165.958 85941.243, 547179.112 859...  
4  POLYGON ((419052.089 80240.803, 419056.851 802...

In [ ]:
# Filter clean shape to ge only the missing rows
df_missing = df1.iloc[missing_values]

# Save as shp
df_missing.to_file(".\\poljine_SHP\\ZV2017_missing_S1")


## 3.2 Add missing values

- the 32 data points were calculated using a slightly smaller buffer (-4 meters)
- they need to be added into the correct position, so the id_17 index will be correct

In [85]:
import pandas as pd
import geopandas as gpd

In [86]:
%%time

# Correct order of data points from clean SHP
gdf = gpd.read_file(".\\poljine_SHP\\ZV2017_d96tm_clean_ids.shp")
# Main data set with missing values
df0 = pd.read_csv(".\\csvs\\ZV2017_extracted_s1-slc.csv")
# Missing values
df1 = pd.read_csv(".\\csvs\\ZV2017_missing_S1_slc_missing.csv")

Wall time: 2min 9s


In [87]:
gdf0 = gdf.drop(columns=["geometry"])
gdf0.head(2)

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
0      0   2537988     2104906     1100  19308.9997        801      GP   
1      1   2537988     2104908     1100  19308.9997        005      GP   

   INTERSECT    POVR_ar  REGIJA  
0          0  57.930456       5  
1          0  30.520097       5

In [88]:
df0.head(2)

GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS OBRAZEC  INTERSECT  \
0   2537988     2104906     1100  19308.9997         801      GP          0   
1   2537988     2104908     1100  19308.9997           5      GP          0   

     POVR_ar  REGIJA  SIG_ASC_VH_04_01  ...  COH_DES_VH_10_22  \
0  57.930456       5          0.012960  ...          0.507982   
1  30.520097       5          0.009604  ...          0.460377   

   COH_DES_VV_10_22  SIG_ASC_VH_10_28  SIG_ASC_VV_10_28  SIG_DES_VH_10_28  \
0          0.524422          0.031385          0.183504          0.027705   
1          0.570117          0.036393          0.204009          0.029903   

   SIG_DES_VV_10_28  COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  \
0          0.180794          0.689269          0.752774          0.580919   
1          0.234453          0.740559          0.745909          0.585225   

   COH_DES_VV_10_28  
0          0.686584  
1          0.734621  

[2 rows x 297 columns]

In [89]:
df1.head(2)

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS OBRAZEC  \
0  46616    759714     2829278     1100   2304.0624         809      PP   
1  47144    759714     2803530     1100   2304.0624         207      GP   

   INTERSECT    POVR_ar  REGIJA  ...  COH_DES_VH_10_22  COH_DES_VV_10_22  \
0          1  23.040624       5  ...          0.464669           0.61585   
1          1  23.040624       5  ...          0.464669           0.61585   

   SIG_ASC_VH_10_28  SIG_ASC_VV_10_28  SIG_DES_VH_10_28  SIG_DES_VV_10_28  \
0          0.011288          0.065602          0.011994          0.086846   
1          0.011288          0.065602          0.011994          0.086846   

   COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
0          0.415111          0.623159          0.363966          0.573629  
1          0.415111          0.623159          0.363966          0.573629  

[2 rows x 298 columns]

In [90]:
df3 = pd.merge(gdf0 , df0, how='left', on='POLJINA_ID', suffixes=('', '_DROP')).filter(regex='^(?!.*_DROP)')
df3

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
0            0   2537988     2104906     1100  19308.9997        801      GP   
1            1   2537988     2104908     1100  19308.9997        005      GP   
2            2   2537988     2104910     1100  19308.9997        203      GP   
3            3   1143420     2105003     1100   8277.3796        801      GP   
4            4   3077991     2652377     1211  14987.7976        100      GP   
...        ...       ...         ...      ...         ...        ...     ...   
664133  664133   1892284     2976726     1300   8760.9728        204      GP   
664134  664134   3588000     2976727     1300   7807.7304        204      GP   
664135  664135   1646627     2976728     1300   7559.2166        204      GP   
664136  664136   3588612     2976729     1300   6120.1665        204      GP   
664137  664137   3588639     2976730     1300   5015.4655        204      GP   

        INTERSECT     POVR_ar  REGIJA  ...  COH_DES_VH_10_22  \
0               0   57.930456       5  ...          0.507982   
1               0   30.520097       5  ...          0.460377   
2               0   26.799442       5  ...          0.433481   
3               0   40.952670       5  ...          0.426090   
4               0  149.877976       4  ...          0.380814   
...           ...         ...     ...  ...               ...   
664133          0   87.609728       2  ...          0.321100   
664134          0   78.077304       2  ...          0.399328   
664135          0   75.592166       2  ...          0.355053   
664136          0   61.201665       2  ...          0.338016   
664137          0   50.154655       2  ...          0.346559   

        COH_DES_VV_10_22  SIG_ASC_VH_10_28  SIG_ASC_VV_10_28  \
0               0.524422          0.031385          0.183504   
1               0.570117          0.036393          0.204009   
2               0.440421          0.018985          0.109362   
3               0.476744          0.020489          0.098561   
4               0.430132          0.023058          0.067281   
...                  ...               ...               ...   
664133          0.413124          0.014662          0.047783   
664134          0.465319          0.012253          0.039134   
664135          0.548025          0.024380          0.065607   
664136          0.299834          0.008648          0.023843   
664137          0.366279          0.007152          0.017471   

        SIG_DES_VH_10_28  SIG_DES_VV_10_28  COH_ASC_VH_10_28  \
0               0.027705          0.180794          0.689269   
1               0.029903          0.234453          0.740559   
2               0.017518          0.079535          0.577215   
3               0.020738          0.099730          0.447147   
4               0.021289          0.080768          0.402380   
...                  ...               ...               ...   
664133          0.046331          0.219224          0.283349   
664134          0.030596          0.129961          0.326760   
664135          0.039934          0.218525          0.421177   
664136          0.027773          0.128125          0.378907   
664137          0.031868          0.168384          0.339139   

        COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
0               0.752774          0.580919          0.686584  
1               0.745909          0.585225          0.734621  
2               0.682407          0.377640          0.448435  
3               0.398452          0.454416          0.428645  
4               0.496410          0.295891          0.316103  
...                  ...               ...               ...  
664133          0.303850          0.255324          0.243495  
664134          0.355066          0.292682          0.346647  
664135          0.368866          0.247987          0.284292  
664136          0.417516          0.312166          0.233042  
664137          0.344201          0.270885      

In [91]:
missing_poljine = df1["POLJINA_ID"]
missing_rows = df3[df3["POLJINA_ID"].isin(missing_poljine.to_numpy())].index
df3.iloc[missing_rows] = df1
df3.iloc[missing_rows]

id_17  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_ SIFRA_KMRS OBRAZEC  \
46616    46616    759714     2829278     1100   2304.0624        809      PP   
47144    47144    759714     2803530     1100   2304.0624        207      GP   
57323    57323   1830184     2937936     1100   2981.0183          6      GP   
62652    62652    140213     2349097     1100   1905.3392        801      GP   
62757    62757    140213     2349179     1100   1905.3392        808      PP   
62763    62763    140213     2349267     1100   1905.3392        112      NP   
93813    93813   1744214     3040677     1300   1703.8102        204      GP   
94575    94575     70106     3027566     1100   1909.6832        112      NP   
97683    97683     70106     3022767     1100   1909.6832         33      GP   
109581  109581    689955     2125959     1300   1252.8071        204      GP   
109906  109906    216061     2120608     1100   1436.8073          5      GP   
115507  115507    725615     2770314     1100   2016.9475        801      GP   
116679  116679    725615     2776223     1100   2016.9475        112      NP   
116680  116680    725615     2776292     1100   2016.9475        814      PP   
135561  135561   2073237     2149591     1300   1209.6132        204      GP   
154766  154766    477474     2175375     1300   3050.3957        204      GP   
204863  204863   2564185     2284026     1300   1460.5567        204      GP   
260352  260352    164113     2370181     1300   2237.1107        204      GP   
291532  291532    703611     2411600     1100   2069.0407        809      GP   
300465  300465   1354754     2445738     1300   1183.6737        204      GP   
326524  326524    887623     2462789     1300   1681.9341        204      GP   
355781  355781     55456     2526326     1100   2886.8635        803      GP   
392932  392932   1710692     3079749     1300   1567.8650        204      GP   
429276  429276    857410     2643890     1100   1906.4174        809      GP   
451339  451339   2111873     2701090     1100   5693.2903         20      GP   
530030  530030    347079     2814063     1300   1427.0662        204      GP   
555758  555758   2516936     2913981     1300   1491.1965        204      GP   
598878  598878    943688     2620489     1300   1378.9806        204      GP   
649259  649259    594871     2962245     1100   3567.5551        222      PP   
650254  650254    594871     2965840     1100   3567.5551        111      NP   
651497  651497    594871     2961541     1100   3567.5551          9      GP   
656362  656362     70106     3092821     1100   1909.6832        807      PP   

        INTERSECT    POVR_ar  REGIJA  ...  COH_DES_VH_10_22  COH_DES_VV_10_22  \
46616           1  23.040624       5  ...          0.464669          0.615850   
47144           1  23.040624       5  ...          0.464669          0.615850   
57323           0  15.616768       5  ...          0.332144          0.521226   
62652           1  19.053392       5  ...          0.425790          0.677048   
62757           1  19.053392       5  ...          0.425790          0.677048   
62763           1  19.053392       5  ...          0.425790          0.677048   
93813           0  17.038102       5  ...          0.443678          0.554677   
94575           1  19.096832       5  ...          0.506524          0.735718   
97683           1  19.096832       5  ...          0.506524          0.735718   
109581          0  12.528071       1  ...          0.289132          0.561356   
109906          0  10.703476       5  ...          0.374557          0.531865   
115507          1  20.169475       5  ...          0.358921          0.368664   
116679          1  20.169475       5  ...          0.358921          0.368664   
116680          1  20.169475       5  ...          0.358921          0.368664   
135561          0  12.096132       2  ...          0.385471          0.451778   
154766          0  30.503957       5  ...          0.284060          0.3

In [93]:
df3.columns.to_list()

['id_17',
 'GERK_PID',
 'POLJINA_ID',
 'RABA_ID',
 'POVR_GERK_',
 'SIFRA_KMRS',
 'OBRAZEC',
 'INTERSECT',
 'POVR_ar',
 'REGIJA',
 'SIG_ASC_VH_04_01',
 'SIG_ASC_VV_04_01',
 'SIG_DES_VH_04_01',
 'SIG_DES_VV_04_01',
 'COH_ASC_VH_04_01',
 'COH_ASC_VV_04_01',
 'COH_DES_VH_04_01',
 'COH_DES_VV_04_01',
 'SIG_ASC_VH_04_07',
 'SIG_ASC_VV_04_07',
 'SIG_DES_VH_04_07',
 'SIG_DES_VV_04_07',
 'COH_ASC_VH_04_07',
 'COH_ASC_VV_04_07',
 'COH_DES_VH_04_07',
 'COH_DES_VV_04_07',
 'SIG_ASC_VH_04_13',
 'SIG_ASC_VV_04_13',
 'SIG_DES_VH_04_13',
 'SIG_DES_VV_04_13',
 'COH_ASC_VH_04_13',
 'COH_ASC_VV_04_13',
 'COH_DES_VH_04_13',
 'COH_DES_VV_04_13',
 'SIG_ASC_VH_04_19',
 'SIG_ASC_VV_04_19',
 'SIG_DES_VH_04_19',
 'SIG_DES_VV_04_19',
 'COH_ASC_VH_04_19',
 'COH_ASC_VV_04_19',
 'COH_DES_VH_04_19',
 'COH_DES_VV_04_19',
 'SIG_ASC_VH_04_25',
 'SIG_ASC_VV_04_25',
 'SIG_DES_VH_04_25',
 'SIG_DES_VV_04_25',
 'COH_ASC_VH_04_25',
 'COH_ASC_VV_04_25',
 'COH_DES_VH_04_25',
 'COH_DES_VV_04_25',
 'SIG_ASC_VH_05_01',
 'SIG_ASC_

In [98]:
# Drop unnecessary columns
df_tidy = df3.drop(columns=df0.iloc[:, 0:9])

df_tidy.tail(3)

id_17  SIG_ASC_VH_04_01  SIG_ASC_VV_04_01  SIG_DES_VH_04_01  \
664135  664135          0.021321          0.064066          0.059747   
664136  664136          0.008370          0.022646          0.029279   
664137  664137          0.006835          0.017040          0.030513   

        SIG_DES_VV_04_01  COH_ASC_VH_04_01  COH_ASC_VV_04_01  \
664135          0.358613          0.303795          0.342158   
664136          0.119966          0.277950          0.339892   
664137          0.165918          0.283985          0.330660   

        COH_DES_VH_04_01  COH_DES_VV_04_01  SIG_ASC_VH_04_07  ...  \
664135          0.334006          0.322785          0.019865  ...   
664136          0.293512          0.292937          0.007930  ...   
664137          0.303999          0.359209          0.008662  ...   

        COH_DES_VH_10_22  COH_DES_VV_10_22  SIG_ASC_VH_10_28  \
664135          0.355053          0.548025          0.024380   
664136          0.338016          0.299834          0.008648   
664137          0.346559          0.366279          0.007152   

        SIG_ASC_VV_10_28  SIG_DES_VH_10_28  SIG_DES_VV_10_28  \
664135          0.065607          0.039934          0.218525   
664136          0.023843          0.027773          0.128125   
664137          0.017471          0.031868          0.168384   

        COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
664135          0.421177          0.368866          0.247987          0.284292  
664136          0.378907          0.417516          0.312166          0.233042  
664137          0.339139          0.344201          0.270885          0.280166  

[3 rows x 289 columns]

In [99]:
%%time

# Save to csv
df_tidy.to_csv(f".\\results_extracted\\ZV2017_extracted_s1-slc.csv", index=False)

Wall time: 5min 56s


# 4. Extracted HAND for 2017 (DONE)

- tidy up the data
- add "id_17" index

In [ ]:
import pandas as pd

In [16]:
%%time

src_pth = f".\\csvs\\ZV2017_extracted_hand.csv"

df0 = pd.read_csv(src_pth)

Wall time: 824 ms


In [17]:
df0

Unnamed: 0  GERK_PID  POLJINA_ID  RABA_ID  POVR_GERK_  SIFRA_KMRS  \
0                0   2537988     2104906     1100  19308.9997         801   
1                1   2537988     2104908     1100  19308.9997           5   
2                2   2537988     2104910     1100  19308.9997         203   
3                3   1143420     2105003     1100   8277.3796         801   
4                4   3077991     2652377     1211  14987.7976         100   
...            ...       ...         ...      ...         ...         ...   
664133      681156   1892284     2976726     1300   8760.9728         204   
664134      681157   3588000     2976727     1300   7807.7304         204   
664135      681158   1646627     2976728     1300   7559.2166         204   
664136      681159   3588612     2976729     1300   6120.1665         204   
664137      681160   3588639     2976730     1300   5015.4655         204   

       OBRAZEC  INTERSECT     POVR_ar  REGIJA  pix_count  hand_mean   hand_std  
0           GP          0   57.930456       5         63   0.429364   0.105657  
1           GP          0   30.520097       5         34   0.379970   0.025663  
2           GP          0   26.799442       5         30   0.428827   0.117456  
3           GP          0   40.952670       5         42   1.999929   0.352233  
4           GP          0  149.877976       4        167   5.813277   3.331927  
...        ...        ...         ...     ...        ...        ...        ...  
664133      GP          0   87.609728       2         92  18.987577  10.833754  
664134      GP          0   78.077304       2         81  23.950169   8.193017  
664135      GP          0   75.592166       2         80  77.466759  24.283596  
664136      GP          0   61.201665       2         65   9.352770   5.397253  
664137      GP          0   50.154655       2         51   9.910322   2.914534  

[664138 rows x 13 columns]

In [18]:
# Drop unnecessary columns
df_tidy = df0.drop(columns=df0.iloc[:, 0:11])
   
# Add the "id_17" index column
df_tidy = df_tidy.reset_index()
df_tidy = df_tidy.rename(columns={"index": "id_17"})

df_tidy.tail(3)

id_17  hand_mean   hand_std
664135  664135  77.466759  24.283596
664136  664136   9.352770   5.397253
664137  664137   9.910322   2.914534

In [19]:
%%time

# Save to csv
df_tidy.to_csv(f".\\results_extracted\\ZV2017_extracted_hand.csv", index=False)

Wall time: 2.82 s


# 5. Extracted indices for 2018 and 2019

- Tidy: rename "index" to "id_18" and "id_19" respectively

## 5.1 Year 2018

There should be 680698 rows in each data set

In [1]:
import pandas as pd

In [72]:
%%time

src_pth = f".\\csvs\\ZV2018_extracted_ndvi.csv"

df0 = pd.read_csv(src_pth)

Wall time: 4.41 s


In [75]:
df = df0.drop(columns=df0.iloc[:, 0:9])
df = df.reset_index()
df = df.rename(columns={"index": "id_18"})
df.tail()

id_17  ndvi_04_01  ndvi_04_06  ndvi_04_11  ndvi_04_16  ndvi_04_21  \
680693  680693         NaN    0.638653    0.611491    0.201033    0.812884   
680694  680694         NaN    0.532274    0.527314    0.213955    0.719594   
680695  680695         NaN    0.362106         NaN         NaN    0.678596   
680696  680696         NaN    0.525302         NaN         NaN    0.702321   
680697  680697         NaN    0.807609         NaN         NaN    0.819370   

        ndvi_04_26  ndvi_05_01  ndvi_05_06  ndvi_05_11  ...  ndvi_09_13  \
680693         NaN    0.849550    0.848288         NaN  ...         NaN   
680694         NaN    0.795405    0.826601         NaN  ...         NaN   
680695         NaN         NaN         NaN    0.377683  ...         NaN   
680696         NaN         NaN         NaN    0.687092  ...    0.293147   
680697         NaN         NaN         NaN    0.840772  ...         NaN   

        ndvi_09_18  ndvi_09_23  ndvi_09_28  ndvi_10_03  ndvi_10_08  \
680693    0.774726         NaN    0.804530         NaN         NaN   
680694    0.799814         NaN    0.806531         NaN         NaN   
680695    0.685558         NaN    0.680902    0.671604         NaN   
680696    0.784263         NaN    0.791756    0.778464         NaN   
680697    0.854973         NaN    0.861242    0.824095         NaN   

        ndvi_10_13  ndvi_10_18  ndvi_10_23  ndvi_10_28  
680693         NaN         NaN    0.741961         NaN  
680694         NaN         NaN    0.681984         NaN  
680695         NaN         NaN         NaN         NaN  
680696         NaN         NaN    0.743478         NaN  
680697         NaN         NaN    0.786069         NaN  

[5 rows x 43 columns]

In [80]:
suffs = ["evi2", "ndvi", "ndwi", "savi"]
for suff in suffs:
    # Open file
    src_pth = f".\\csvs\\ZV2018_extracted_{suff}.csv"
    df0 = pd.read_csv(src_pth)
    
    # Drop unnecessary columns
    df_tidy = df0.drop(columns=df0.iloc[:, 0:9])  # KMRS data

    # Add the "id_18" index column
    df_tidy = df_tidy.reset_index()
    df_tidy = df_tidy.rename(columns={"index": "id_18"})

    print(df_tidy.tail(1))
    
    # Save to csv
    df_tidy.to_csv(f".\\results_extracted\\ZV2018_extracted_{suff}.csv", index=False)

         id_18  evi2_04_01  evi2_04_06  evi2_04_11  evi2_04_16  evi2_04_21  \
680697  680697         NaN    0.514897         NaN         NaN    0.596118   

        evi2_04_26  evi2_05_01  evi2_05_06  evi2_05_11  ...  evi2_09_13  \
680697         NaN         NaN         NaN    0.657612  ...         NaN   

        evi2_09_18  evi2_09_23  evi2_09_28  evi2_10_03  evi2_10_08  \
680697    0.678097         NaN    0.701305    0.650388         NaN   

        evi2_10_13  evi2_10_18  evi2_10_23  evi2_10_28  
680697         NaN         NaN    0.596261         NaN  

[1 rows x 43 columns]
         id_18  ndvi_04_01  ndvi_04_06  ndvi_04_11  ndvi_04_16  ndvi_04_21  \
680697  680697         NaN    0.807609         NaN         NaN     0.81937   

        ndvi_04_26  ndvi_05_01  ndvi_05_06  ndvi_05_11  ...  ndvi_09_13  \
680697         NaN         NaN         NaN    0.840772  ...         NaN   

        ndvi_09_18  ndvi_09_23  ndvi_09_28  ndvi_10_03  ndvi_10_08  \
680697    0.854973         NaN    0.

## 5.2 Year 2019

There should be 626061 rows in each data set

In [1]:
import pandas as pd

In [82]:
%%time

src_pth = f".\\csvs\\KMRS2019_extracted_ndvi.csv"

df0 = pd.read_csv(src_pth)

Wall time: 4.07 s


In [83]:
df = df0.drop(columns=df0.iloc[:, 0:4])
df = df.reset_index()
df = df.rename(columns={"index": "id_19"})
df.tail()

id_19  ndvi_04_01  ndvi_04_06  ndvi_04_11  ndvi_04_16  ndvi_04_21  \
626057  626057         NaN         NaN         NaN    0.602498    0.665913   
626058  626058         NaN         NaN         NaN    0.643462    0.663859   
626059  626059         NaN         NaN         NaN    0.363948    0.391153   
626060  626060         NaN         NaN         NaN    0.745655    0.784228   
626061  626061         NaN         NaN         NaN    0.759347    0.793245   

        ndvi_04_26  ndvi_05_01  ndvi_05_06  ndvi_05_11  ...  ndvi_09_08  \
626057         NaN    0.742322         NaN         NaN  ...         NaN   
626058         NaN    0.734341         NaN         NaN  ...         NaN   
626059         NaN    0.541706         NaN         NaN  ...         NaN   
626060         NaN    0.831408         NaN         NaN  ...         NaN   
626061         NaN    0.854751    0.928058         NaN  ...         NaN   

        ndvi_09_13  ndvi_09_18  ndvi_09_23  ndvi_09_28  ndvi_10_08  \
626057    0.848981    0.356762         NaN         NaN    0.829991   
626058    0.741071         NaN         NaN         NaN    0.764683   
626059    0.544828         NaN         NaN         NaN    0.496529   
626060    0.838207         NaN         NaN         NaN    0.821985   
626061    0.860950         NaN         NaN         NaN    0.787622   

        ndvi_10_13  ndvi_10_18  ndvi_10_23  ndvi_10_28  
626057         NaN         NaN         NaN         NaN  
626058         NaN         NaN         NaN         NaN  
626059         NaN         NaN         NaN         NaN  
626060         NaN         NaN         NaN         NaN  
626061         NaN         NaN         NaN    0.201204  

[5 rows x 43 columns]

In [84]:
suffs = ["evi2", "ndvi", "ndwi", "savi"]
for suff in suffs:
    # Open file
    src_pth = f".\\csvs\\KMRS2019_extracted_{suff}.csv"
    df0 = pd.read_csv(src_pth)
    
    # Drop unnecessary columns
    df_tidy = df0.drop(columns=df0.iloc[:, 0:4])  # KMRS data

    # Add the "id_18" index column
    df_tidy = df_tidy.reset_index()
    df_tidy = df_tidy.rename(columns={"index": "id_19"})

    print(df_tidy.tail(1))
    
    # Save to csv
    df_tidy.to_csv(f".\\results_extracted\\KMRS2019_extracted_{suff}.csv", index=False)

         id_19  evi2_04_01  evi2_04_06  evi2_04_11  evi2_04_16  evi2_04_21  \
626061  626061         NaN         NaN         NaN    0.517521    0.579275   

        evi2_04_26  evi2_05_01  evi2_05_06  evi2_05_11  ...  evi2_09_08  \
626061         NaN    0.660095    0.496133         NaN  ...         NaN   

        evi2_09_13  evi2_09_18  evi2_09_23  evi2_09_28  evi2_10_08  \
626061    0.632914         NaN         NaN         NaN     0.55424   

        evi2_10_13  evi2_10_18  evi2_10_23  evi2_10_28  
626061         NaN         NaN         NaN    0.106855  

[1 rows x 43 columns]
         id_19  ndvi_04_01  ndvi_04_06  ndvi_04_11  ndvi_04_16  ndvi_04_21  \
626061  626061         NaN         NaN         NaN    0.759347    0.793245   

        ndvi_04_26  ndvi_05_01  ndvi_05_06  ndvi_05_11  ...  ndvi_09_08  \
626061         NaN    0.854751    0.928058         NaN  ...         NaN   

        ndvi_09_13  ndvi_09_18  ndvi_09_23  ndvi_09_28  ndvi_10_08  \
626061     0.86095         NaN      

# 6. Extracted s2-rgbn for 2018 and 2019

- Don't rename index
- Only remove unneeded columns

## 6.1 Year 2018

There should be 680698 rows in each data set

In [1]:
import pandas as pd

In [2]:
%%time

src_pth = f".\\csvs\\aZV2018_d96tm_clean_ids_rgbn.csv"

df0 = pd.read_csv(src_pth)

Wall time: 2.41 s


In [4]:
df0.tail(2)

id_18  GERK_PID  RABA_ID  POVR_GERK_  POLJINA_ID  SIFRA_KMRS OBRAZEC  \
114626  680695   5944379     1211   5816.1295     4079093         100      GP   
114627  680696   4836379     1100   6689.5872     4079099          20      GP   

          POVR_ar  INTERSECT  blue_04_01  ...  red_10_18  nir_10_18  \
114626  58.162621          0         NaN  ...        NaN        NaN   
114627  16.568304          0         NaN  ...        NaN        NaN   

        blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
114626         NaN          NaN        NaN        NaN         NaN   
114627    0.070877     0.083625    0.06289   0.359624         NaN   

        green_10_28  red_10_28  nir_10_28  
114626          NaN        NaN        NaN  
114627          NaN        NaN        NaN  

[2 rows x 177 columns]

In [6]:
df_tidy = df0.drop(columns=df0.iloc[:, 1:9])
df_tidy.tail()

id_18  blue_04_01  green_04_01  red_04_01  nir_04_01  blue_04_06  \
114623  680666         NaN          NaN        NaN        NaN    0.104746   
114624  680669         NaN          NaN        NaN        NaN    0.085881   
114625  680694         NaN          NaN        NaN        NaN    0.074788   
114626  680695         NaN          NaN        NaN        NaN    0.081931   
114627  680696         NaN          NaN        NaN        NaN    0.071619   

        green_04_06  red_04_06  nir_04_06  blue_04_11  ...  red_10_18  \
114623     0.128282   0.130777   0.296457         NaN  ...   0.098429   
114624     0.098508   0.099785   0.286889    0.062532  ...        NaN   
114625     0.091805   0.100300   0.311365    0.068055  ...        NaN   
114626     0.096172   0.112594   0.208005         NaN  ...        NaN   
114627     0.085058   0.086243   0.254237         NaN  ...        NaN   

        nir_10_18  blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
114623   0.151888         NaN          NaN        NaN        NaN    0.265292   
114624        NaN    0.072267     0.090111   0.064208   0.322389         NaN   
114625        NaN    0.072361     0.093147   0.080779   0.363520         NaN   
114626        NaN         NaN          NaN        NaN        NaN         NaN   
114627        NaN    0.070877     0.083625   0.062890   0.359624         NaN   

        green_10_28  red_10_28  nir_10_28  
114623     0.234878   0.199835   0.295654  
114624          NaN        NaN        NaN  
114625          NaN        NaN        NaN  
114626          NaN        NaN        NaN  
114627          NaN        NaN        NaN  

[5 rows x 169 columns]

In [7]:
# Save to csv
df_tidy.to_csv(f".\\results_extracted\\ZV2018_extracted_s2-rgbn.csv", index=False)

## 6.2 Year 2019

There should be 626061 rows in each data set

In [8]:
import pandas as pd

In [9]:
%%time

src_pth = f".\\csvs\\aKMRS2019_d96tm_clean_ids_rgbn.csv"

df0 = pd.read_csv(src_pth)

Wall time: 2.36 s


In [10]:
df0.tail(2)

id_19  GERK_PID  POLJINA_ID  SIFRA_KMRS  blue_04_01  green_04_01  \
111886  626043   3703609     5072640         699    0.063915      0.08004   
111887  626054   6097344     5079147           5         NaN          NaN   

        red_04_01  nir_04_01  blue_04_06  green_04_06  ...  red_10_18  \
111886   0.084677   0.263924         NaN          NaN  ...   0.056515   
111887        NaN        NaN    0.050041     0.071638  ...   0.077625   

        nir_10_18  blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
111886   0.283947         NaN          NaN        NaN        NaN    0.147926   
111887   0.199180         NaN          NaN        NaN        NaN         NaN   

        green_10_28  red_10_28  nir_10_28  
111886     0.163969    0.15613   0.277193  
111887          NaN        NaN        NaN  

[2 rows x 172 columns]

In [12]:
df_tidy = df0.drop(columns=df0.iloc[:, 1:4])
df_tidy.tail()

id_19  blue_04_01  green_04_01  red_04_01  nir_04_01  blue_04_06  \
111883  626020         NaN          NaN        NaN        NaN    0.243413   
111884  626030         NaN          NaN        NaN        NaN         NaN   
111885  626042    0.089221     0.117534   0.148341   0.242135         NaN   
111886  626043    0.063915     0.080040   0.084677   0.263924         NaN   
111887  626054         NaN          NaN        NaN        NaN    0.050041   

        green_04_06  red_04_06  nir_04_06  blue_04_11  ...  red_10_18  \
111883     0.245845   0.245884    0.37788         NaN  ...   0.137803   
111884          NaN        NaN        NaN         NaN  ...        NaN   
111885          NaN        NaN        NaN         NaN  ...   0.102181   
111886          NaN        NaN        NaN         NaN  ...   0.056515   
111887     0.071638   0.090383    0.31772         NaN  ...   0.077625   

        nir_10_18  blue_10_23  green_10_23  red_10_23  nir_10_23  blue_10_28  \
111883   0.231506         NaN          NaN        NaN        NaN         NaN   
111884        NaN         NaN          NaN        NaN        NaN         NaN   
111885   0.275552         NaN          NaN        NaN        NaN    0.185409   
111886   0.283947         NaN          NaN        NaN        NaN    0.147926   
111887   0.199180         NaN          NaN        NaN        NaN         NaN   

        green_10_28  red_10_28  nir_10_28  
111883          NaN        NaN        NaN  
111884          NaN        NaN        NaN  
111885     0.201066   0.195049   0.296492  
111886     0.163969   0.156130   0.277193  
111887          NaN        NaN        NaN  

[5 rows x 169 columns]

In [13]:
%%time
    
# Save to csv
df_tidy.to_csv(f".\\results_extracted\\KMRS2019_extracted_s2-rgbn.csv", index=False)

Wall time: 19.5 s


# 7. Extracted S1-slc for 2018 (DONE)

Clean unneeded columns. There are some NaN values that will be delt with in monthly interpolation.

In [14]:
%%time
import pandas as pd

s1_pth = f".\\csvs\\bZV2018_d96tm_clean_ids_slc.csv"

df1 = pd.read_csv(s1_pth)

Wall time: 6.91 s


In [27]:
df1.head(2)

id_18  GERK_PID  RABA_ID  POVR_GERK_  POLJINA_ID  SIFRA_KMRS OBRAZEC  \
0      1   4896286     1100   1187.8658     3280115         405      GP   
1      6   1083878     1100   7093.8082     3094837           5      GP   

     POVR_ar  INTERSECT  SIG_ASC_VH_04_01  ...  COH_DES_VH_10_22  \
0  11.878907          0          0.021503  ...          0.243379   
1  34.573176          1          0.015046  ...          0.234710   

   COH_DES_VV_10_22  SIG_ASC_VH_10_28  SIG_ASC_VV_10_28  SIG_DES_VH_10_28  \
0          0.296769          0.018704          0.183421          0.027247   
1          0.273542          0.029935          0.222554          0.030716   

   SIG_DES_VV_10_28  COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  \
0          0.176039          0.639720          0.718057          0.285368   
1          0.360910          0.526789          0.756344          0.432395   

   COH_DES_VV_10_28  
0          0.596231  
1          0.496262  

[2 rows x 297 columns]

In [29]:
# Drop unnecessary columns
df_tidy = df1.drop(columns=df1.iloc[:, 1:9])

df_tidy.tail(3)

id_18  SIG_ASC_VH_04_01  SIG_ASC_VV_04_01  SIG_DES_VH_04_01  \
114625  680694          0.015966          0.055539          0.079303   
114626  680695          0.012090          0.073179          0.024907   
114627  680696          0.012670          0.077836          0.020482   

        SIG_DES_VV_04_01  COH_ASC_VH_04_01  COH_ASC_VV_04_01  \
114625          0.316503          0.422478          0.352962   
114626          0.161575          0.400080          0.641991   
114627          0.078194          0.292865          0.471435   

        COH_DES_VH_04_01  COH_DES_VV_04_01  SIG_ASC_VH_04_07  ...  \
114625          0.345540          0.331329          0.019539  ...   
114626          0.505537          0.762077          0.014889  ...   
114627          0.308208          0.293562          0.014387  ...   

        COH_DES_VH_10_22  COH_DES_VV_10_22  SIG_ASC_VH_10_28  \
114625          0.242983          0.352986          0.021054   
114626          0.301089          0.375630          0.016575   
114627          0.335093          0.397810          0.018259   

        SIG_ASC_VV_10_28  SIG_DES_VH_10_28  SIG_DES_VV_10_28  \
114625          0.040198          0.093014          1.096049   
114626          0.073304          0.023659          0.099270   
114627          0.072972          0.030438          0.098792   

        COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
114625          0.321002          0.232757          0.422077          0.376429  
114626          0.314066          0.465852          0.377464          0.456217  
114627          0.333229          0.424934          0.307876          0.416933  

[3 rows x 289 columns]

In [30]:
%%time

# Save to csv
df_tidy.to_csv(f".\\results_extracted\\ZV2018_extracted_s1-slc.csv", index=False)

Wall time: 55.5 s


# 8. Extracted S1-slc for 2019 (DONE)

Clean unneeded columns. There are some NaN values that will be delt with in monthly interpolation.

In [1]:
%%time
import pandas as pd

s1_pth = f".\\csvs\\bKMRS2019_d96tm_clean_ids_slc.csv"

df1 = pd.read_csv(s1_pth)

Wall time: 14 s


In [2]:
df1.head(2)

id_19  GERK_PID  POLJINA_ID  SIFRA_KMRS  SIG_ASC_VH_04_01  \
0      9   3366021     4088919         100          0.020135   
1     12   3365931     4088922         100          0.025754   

   SIG_ASC_VV_04_01  SIG_DES_VH_04_01  SIG_DES_VV_04_01  COH_ASC_VH_04_01  \
0          0.089681          0.025359          0.113961          0.207774   
1          0.110617          0.029655          0.081080          0.278651   

   COH_ASC_VV_04_01  ...  COH_DES_VH_10_22  COH_DES_VV_10_22  \
0          0.317133  ...          0.258645          0.318927   
1          0.323784  ...          0.375712          0.288501   

   SIG_ASC_VH_10_28  SIG_ASC_VV_10_28  SIG_DES_VH_10_28  SIG_DES_VV_10_28  \
0          0.034445          0.104209          0.024351          0.119893   
1          0.023414          0.083092          0.029539          0.109738   

   COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
0          0.323609          0.297022          0.271154          0.241363  
1          0.396411          0.314191          0.281515          0.288947  

[2 rows x 292 columns]

In [4]:
# Drop unnecessary columns
df_tidy = df1.drop(columns=df1.iloc[:, 1:4])

df_tidy.tail(3)

id_19  SIG_ASC_VH_04_01  SIG_ASC_VV_04_01  SIG_DES_VH_04_01  \
111885  626042          0.022681          0.121705          0.011598   
111886  626043          0.053472          0.234808          0.022507   
111887  626054          0.013006          0.072728          0.007915   

        SIG_DES_VV_04_01  COH_ASC_VH_04_01  COH_ASC_VV_04_01  \
111885          0.070266          0.271903          0.423134   
111886          0.059021          0.190250          0.535585   
111887          0.038693          0.350579          0.418133   

        COH_DES_VH_04_01  COH_DES_VV_04_01  SIG_ASC_VH_04_07  ...  \
111885          0.262441          0.391141          0.053226  ...   
111886          0.262182          0.270252          0.058037  ...   
111887          0.279014          0.421297          0.015509  ...   

        COH_DES_VH_10_22  COH_DES_VV_10_22  SIG_ASC_VH_10_28  \
111885          0.286680          0.352370          0.038557   
111886          0.288292          0.326793          0.061846   
111887          0.339224          0.439304          0.030884   

        SIG_ASC_VV_10_28  SIG_DES_VH_10_28  SIG_DES_VV_10_28  \
111885          0.164237          0.023545          0.084672   
111886          0.269675          0.018982          0.060183   
111887          0.207175          0.027955          0.094833   

        COH_ASC_VH_10_28  COH_ASC_VV_10_28  COH_DES_VH_10_28  COH_DES_VV_10_28  
111885          0.272378          0.357639          0.297521          0.326979  
111886          0.319101          0.346832          0.256712          0.321693  
111887          0.377051          0.531453          0.315086          0.367979  

[3 rows x 289 columns]

In [5]:
%%time

# Save to csv
df_tidy.to_csv(f".\\results_extracted\\KMRS2019_extracted_s1-slc.csv", index=False)

Wall time: 1min 13s
